api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationSearchAdviserTest.java

```diff
@@ -0,0 +1,446 @@
+        authenticateUserWithLocationProperty();
+        assertAuthenticatedNormalUser();
+
+        List<String> result1 = getUuidsOfLocations(locationService.getLocations(searchQuery));
+        assertEquals(1, result1.size());
+        assertEquals(location3Uuid, result1.get(0));
+
+        List<String> result2 = getUuidsOfLocations(locationService.getLocations(searchQuery, null, null, true, null, null));
+        assertEquals(1, result2.size());
+        assertEquals(location3Uuid, result1.get(0));
+    }
+
+    @Test
+    public void getLocations_nonAdminWithoutLocationPropertyShouldGetOnlySessionLocation() {
+        authenticateUserWithoutLocationProperty();
+        assertAuthenticatedNormalUser();
+
+        setSessionLocationID(2);
+
+        List<String> result1 = getUuidsOfLocations(locationService.getLocations(searchQuery));
+        assertEquals(1, result1.size());
+        assertEquals(location2Uuid, result1.get(0));
+        List<String> result2 = getUuidsOfLocations(locationService.getLocations(searchQuery, null, null, true, null, null));
+        assertEquals(1, result2.size());
+        assertEquals(location2Uuid, result1.get(0));
+
+        setSessionLocationID(4);
+
+        List<String> result3 = getUuidsOfLocations(locationService.getLocations(searchQuery));
+        assertEquals(0, result3.size());
+        List<String> result4 = getUuidsOfLocations(locationService.getLocations(searchQuery, null, null, true, null, null));
+        assertEquals(1, result4.size());
+        assertEquals(location4Uuid, result4.get(0));
+    }
+
+    @Test
+    public void getLocationsByTag_adminShouldGetAllLocationsWithTag() {
+        authenticateAdmin();
+        assertAuthenticatedAdmin();
+        LocationTag tag = locationService.getLocationTagByName(tagName);
+
+        List<String> result = getUuidsOfLocations(locationService.getLocationsByTag(tag));
+        assertEquals(2, result.size());
+        assertBothCollectionsAreEqual(locationsWithTagUuids, result);
+    }
+
+    @Test
+    public void getLocationByTag_nonAdminWithLocationPropertyShouldGetOnlyAccessLocation() {
+        authenticateUserWithLocationProperty();
+        assertAuthenticatedNormalUser();
+        LocationTag tag = locationService.getLocationTagByName(tagName);
+
+        List<String> result = getUuidsOfLocations(locationService.getLocationsByTag(tag));
+        assertEquals(1, result.size());
+        assertEquals(location3Uuid, result.get(0));
+    }
+
+    @Test
+    public void getLocationByTag_nonAdminWithoutLocationPropertyShouldGetOnlySessionLocation() {
+        authenticateUserWithoutLocationProperty();
+        assertAuthenticatedNormalUser();
+        LocationTag tag = locationService.getLocationTagByName(tagName);
+
+        setSessionLocationID(1);
+
+        List<String> result1 = getUuidsOfLocations(locationService.getLocationsByTag(tag));
+        assertEquals(1, result1.size());
+        assertEquals(location1Uuid, result1.get(0));
```

Comment:
Here, You have set SessionLocationId as 1, and then check for location1Uuid. How can we ensure that, both are related to one location object? 
So initialize one location object with all properties, and use that object to get ids, get uuids, etc (e.g.g location1.getId()).

Reply:
Do this for all test cases here. Reduce the multiple variables usages among test cases.